### PR TITLE
Feature/updates

### PR DESCRIPTION
**Pull request by iruzevic**
Various changes:
- BrowserSync related to this PR https://github.com/infinum/eightshift-frontend-libs/pull/267
- Webpack Dynamic Import - https://github.com/infinum/eightshift-docs/issues/73
- Dashed component name issue - https://github.com/infinum/eightshift-docs/issues/72
- Sharing assets across languages - https://github.com/infinum/eightshift-docs/issues/69
- Adding reusable blocks chapter - https://github.com/infinum/eightshift-docs/issues/63
- Adding chapter for better usage of components - https://github.com/infinum/eightshift-docs/issues/68
- Simpe vs compoind blocks - https://github.com/infinum/eightshift-docs/issues/67
- New component name - https://github.com/infinum/eightshift-docs/issues/71

### CODE REVIEW

**Review by gabriel-glo — APPROVED** [review]

File: website/docs/basics/blocks-component-in-block.md
Comment (volfkarlo) [inline website/docs/basics/blocks-component-in-block.md:321]:
```suggestion
We created this cool way of importing attributes and loading components in your blocks or other components. However, that doesn't mean that you can't have instances where you have to load components manually. In that case, you have to provide all the attributes that you want to use manually. The attributes that you didn't provide will not be magically set from the manifest, but you can make your life easier and map it like this.
```

Sorry, I'm replacing _must_ with _have to_ because it sounds softer but it's giving an understanding that it is necessary. It's not a must(pun intended) but I believe it's a bit nicer to read. 🙂

File: website/docs/basics/block-manifest.md
Comment (IvanGrginovInf) [inline website/docs/basics/block-manifest.md:91]:
```suggestion
As described before we use `manifest.json` to share stuff across PHP / JS / SCSS so you can easily see its power.
```

File: website/docs/basics/block-manifest.md
Comment (IvanGrginovInf) [inline website/docs/basics/block-manifest.md:93]:
```suggestion
For example, you can use `manifest.json` to store SVG files that you will use on the frontend (PHP) and in the editor / backend (JS) files. Or you can store block options so it is easier to find and add items.
```

File: website/docs/basics/blocks-intro.md
Comment (volfkarlo) [inline website/docs/basics/blocks-intro.md:216]:
```suggestion
* Make your names as generic as possible for better reusability, but specific enough to fully understand what the block/component is used for.
```

I would suggest to add this part. Not sure if everyone agrees but for me it seems useful.

File: website/docs/basics/blocks-component-in-block.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-component-in-block.md:321]:
```suggestion
We created this cool way of importing attributes and loading components in your blocks or other components. Even so, you might have instances where you will have to render components manually. In that case, you must provide all attributes that your component uses manually. Attributes that you didn't provide will not be magically set from the manifest but you can make your life easier and map it like this.
```

File: website/docs/basics/blocks-intro.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-intro.md:198]: Can't leave a comment on that line but there's a heading right above it:

> ### Why is this my blocks folder called custom and not blocks, for example?

I'd change that to: 

> ### Why is my blocks folder called `custom` and not `blocks`, for example?

File: website/docs/basics/blocks-intro.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-intro.md:204]:
```suggestion
**Simple block** is a block that is used just as is and provides a small isolated functionality. For example: heading, paragraph, button, etc.
```

File: website/docs/basics/blocks-intro.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-intro.md:206]:
```suggestion
**Compound block** is a block built from multiple components. For example: card, featured posts, etc
```

File: website/docs/basics/blocks-intro.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-intro.md:215]:
```suggestion
* Always try to name your block based on **what they are**, rather than **what they will be used for**. For example, if you have a component card for custom post-type `books`, you should never call this component `card-book`. Instead, use a more generic name like `card-product`.
```

File: website/docs/basics/blocks-intro.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-intro.md:218]:
```suggestion
* When naming your attributes ask yourself: "is this attribute going to be used in any other way?". For example, you have an attribute for adding font-weight: bold to your text. You can create an attribute called font-weight and set it as a `boolean` type and that will be ok if you have only one font-weight. A better way would be to put it as a string and provide a `SelectControl` component if there is any possibility that in the future you will have additional font-weight.
```

File: website/docs/basics/blocks-intro.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-intro.md:219]:
```suggestion
* Name booleans positively as a question. Example: `isValid`, `isLoading`, `isComplete`.
```

File: website/docs/basics/blocks-intro.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-intro.md:220]: > We use short words even if it doesn’t provide full insight about the code which is a bad thing to do.

Do we need to open an issue for this? :D

File: website/docs/basics/blocks-intro.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-intro.md:224]:
```suggestion
* You might not be the only person working on your particular project in the future. Think about the next person. Your naming will provide an insight into the process and project.
```

File: website/docs/basics/blocks-patterns.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-patterns.md:29]: Maybe add a paragraph about "why you'd want to do this" ("remove all core patterns):

_Why? If you don't use core blocks, core patterns won't work. Don't confuse your users and remove them (but feel free to add new patterns based on your own custom blocks!)_

File: website/docs/basics/blocks-reusable.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-reusable.md:38]: Notes about reusable blocks, they're not native WP search friendly (as in the content you only have the WP block comment markup that there's a specific reusable block there). So if you use a lot of these in the content of a page / post, it might affect your search relevancy.

File: website/docs/basics/blocks-special-use-cases.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-special-use-cases.md:90]:
```suggestion
## How to only use my custom blocks
```

File: website/docs/basics/blocks-special-use-cases.md
Comment (IvanGrginovInf) [inline website/docs/basics/blocks-special-use-cases.md:104]:
```suggestion
You create a new callback method for the `allowed_block_types` filter. There you can allow all your custom blocks + any other 3rd party / core block you want.
```

File: website/docs/basics/browser-sync.md
Comment (IvanGrginovInf) [inline website/docs/basics/browser-sync.md:38]:
```suggestion
By default BrowserSync does not work with https urls but if you use HTTPS in your local environment you can simply provide additional key to the Webpack config to make it work.
```

File: website/docs/basics/dynamic-import.md
Comment (IvanGrginovInf) [inline website/docs/basics/dynamic-import.md:20]:
```suggestion
In a nutshell, you load all your code using JavaScript promises and Webpack splits all these promises into separate file chinks. Once the condition to resolve the promise is set (generally we check if the selector is present in the DOM), Webpack automatically injects your chunk in the DOM and loads it up. This is why you will see in your public folder a bunch of smaller files called (`0.js`, `1.js`, `2.js`, etc).
```

File: website/docs/basics/dynamic-import.md
Comment (IvanGrginovInf) [inline website/docs/basics/dynamic-import.md:22]:
```suggestion
_You don't need to think about this. Webpack just does it for you._
```

**Review by volfkarlo — APPROVED** [review]
All of my comments are suggestions, and for that reason I'm approving with few minor comments.

File: website/docs/basics/dynamic-import.md
Comment (IvanGrginovInf) [inline website/docs/basics/dynamic-import.md:83]:
```suggestion
  // This is the important part. Because of it, your code will only be imported when there's an element on the page that uses it.
```

File: website/docs/basics/dynamic-import.md
Comment (IvanGrginovInf) [inline website/docs/basics/dynamic-import.md:88]:
```suggestion
      // Load a class for all sliders
```

Because (at least specifically for the carousel in this example) it's doesn't initialize the class for each slider on the page but rather (I assume) it initializes all of them in the `init()` method. (If not, we need a look here for each Carousel on the page)

**Review by IvanGrginovInf — CHANGES_REQUESTED** [review]
Added a few suggestions, otherwise awesome work!

File: website/docs/basics/block-manifest.md
Comment (dingo-d) [reply website/docs/basics/block-manifest.md:91]: Also: 

```As described before, we use `manifest.json` to share stuff across PHP / JS / SCSS so you can easily see its power.```

The comma after before word.

File: website/docs/basics/block-manifest.md
Comment (dingo-d) [inline website/docs/basics/block-manifest.md:95]:
```suggestion
Let's see how you would share SVG icon across multiple languages:
```

File: website/docs/basics/blocks-intro.md
Comment (dingo-d) [inline website/docs/basics/blocks-intro.md:212]:
```suggestion
Yes, naming is hard, and no matter how long your development experience is you will always struggle with names for your components, block, files, variables, functions, etc.
```

File: website/docs/basics/blocks-intro.md
Comment (dingo-d) [reply website/docs/basics/blocks-intro.md:215]: The `product` also seems specific, no? Maybe use `card-item`? Or something more general. Product is used if you are using some kind of web stores.

File: website/docs/basics/blocks-intro.md
Comment (dingo-d) [reply website/docs/basics/blocks-intro.md:220]: Yeah, kinda goes against what you are saying in this chapter 😅

File: website/docs/basics/blocks-intro.md
Comment (dingo-d) [inline website/docs/basics/blocks-intro.md:225]: Except when naming components 👀  🙈

File: website/docs/basics/blocks-reusable.md
Comment (dingo-d) [reply website/docs/basics/blocks-reusable.md:38]: Didn't know this. Cool info. Should be added as a useful note. But the idea is kinda to use them as partials, no?

Odd that they won't be searchable 🤔

File: website/docs/basics/blocks-special-use-cases.md
Comment (dingo-d) [inline website/docs/basics/blocks-special-use-cases.md:115]:
```suggestion
  * Filter which blocks are displayed in the block editor.
```

File: website/docs/basics/blocks-special-use-cases.md
Comment (dingo-d) [inline website/docs/basics/blocks-special-use-cases.md:123]: `$allowedBlockTypes` can be `boolean` or `array` so there should be a check before doing any merging

```
if (is_array($allowedBlockTypes)) {
	$allowedBlocks = array_merge(parent::getAllBlocksList($allowedBlockTypes, $post), $allowedBlockTypes);
} else {
	$allowedBlocks = parent::getAllBlocksList($allowedBlockTypes, $post);
}
// additional logic here.

// Remove duplicates.
return array_keys(array_flip($allowedBlocks));
``` 

I got burned by this on one occasion.

File: website/docs/basics/blocks-special-use-cases.md
Comment (dingo-d) [inline website/docs/basics/blocks-special-use-cases.md:146]:
```suggestion
  * Filter which blocks are allowed in what post type.
```

File: website/docs/basics/blocks-special-use-cases.md
Comment (dingo-d) [inline website/docs/basics/blocks-special-use-cases.md:170]: Not sure if the above check is needed here as well 🤷🏼‍♂️

File: website/docs/basics/browser-sync.md
Comment (dingo-d) [inline website/docs/basics/browser-sync.md:36]: I think TLS is used nowadays 🤔

File: website/docs/basics/dynamic-import.md
Comment (dingo-d) [reply website/docs/basics/dynamic-import.md:20]: Typo: `chinks` -> `chunks`

File: website/docs/basics/dynamic-import.md
Comment (dingo-d) [inline website/docs/basics/dynamic-import.md:35]: Not 100% what you meant here.

File: website/docs/basics/dynamic-import.md
Comment (dingo-d) [inline website/docs/basics/dynamic-import.md:71]:
```suggestion
If you are sure you will have only one instance of this feature on one-page use the following code in your `index.js` file.
```

File: website/docs/basics/dynamic-import.md
Comment (dingo-d) [inline website/docs/basics/dynamic-import.md:35]:
```suggestion
This is a version where you have multiple instances of this feature on one page. If you are creating a feature for your block/component you will use the following example.
```

File: website/docs/basics/dynamic-import.md
Comment (dingo-d) [inline website/docs/basics/dynamic-import.md:120]:
```suggestion
        //...
```

File: website/docs/basics/dynamic-import.md
Comment (dingo-d) [inline website/docs/basics/dynamic-import.md:125]:
```suggestion
        // ...
        navigation: {
          nextEl: this.nextElement,
          prevEl: this.prevElement,
        },
        // ...
```

**Review by dingo-d — CHANGES_REQUESTED** [review]
Left some comments, and some minor typo fixes. Awesome work 🙂  👍🏼

File: website/docs/basics/blocks-intro.md
Comment (iruzevic) [reply website/docs/basics/blocks-intro.md:220]: fixed

File: website/docs/basics/blocks-intro.md
Comment (iruzevic) [reply website/docs/basics/blocks-intro.md:225]: removed

File: website/docs/basics/blocks-patterns.md
Comment (iruzevic) [reply website/docs/basics/blocks-patterns.md:29]: done

File: website/docs/basics/blocks-reusable.md
Comment (iruzevic) [reply website/docs/basics/blocks-reusable.md:38]: they are stored in admin as a new post type and has internal connecting to the usage in post/page. So this refers to the native WP search.

Comment (iruzevic) [reply website/docs/basics/blocks-reusable.md:38]: added a note

File: website/docs/basics/blocks-special-use-cases.md
Comment (iruzevic) [reply website/docs/basics/blocks-special-use-cases.md:170]: will test

**Pull request merged**
